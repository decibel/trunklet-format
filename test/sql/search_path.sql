\set ECHO none
\i test/pgxntool/setup.sql
\i test/deps.sql

SELECT plan(
    0

    +2 -- verify search_path
);

SAVEPOINT a;
CREATE TEMP TABLE before AS
  SELECT current_setting('search_path')
;

SELECT pg_temp._create();

CREATE TEMP TABLE after AS
  SELECT current_setting('search_path')
;

-- Forcibly reset it so we know tap will work
SET search_path=public,tap;

SELECT is(
  (SELECT current_setting FROM after)
  , (SELECT current_setting FROM before)
  , 'verify search_path has not changed without cascade'
);

ROLLBACK TO a;
SET LOCAL client_min_messages = WARNING;
CREATE TEMP TABLE before AS
  SELECT current_setting('search_path')
;

SELECT pg_temp._create();

CREATE TEMP TABLE after AS
  SELECT current_setting('search_path')
;

-- Forcibly reset it so we know tap will work
SET search_path=public,tap;

SELECT is(
  (SELECT current_setting FROM after)
  , (SELECT current_setting FROM before)
  , 'verify search_path has not changed with cascade'
);


\i test/pgxntool/finish.sql

-- vi: expandtab ts=2 sw=2
