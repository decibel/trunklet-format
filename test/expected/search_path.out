\set ECHO none
1..2

ok 1 - verify search_path has not changed without cascade

ok 2 - verify search_path has not changed with cascade
# TRANSACTION INTENTIONALLY LEFT OPEN!
